WdtTransferRequest.h

```diff
@@ -35,6 +35,12 @@ struct WdtFileInfo {
   /// Whether read should be done using o_direct. If fd is set, this flag will
   /// be set automatically to match the fd open mode
   bool directReads{false};
+  /// File permission in decimal. When smaller than oct777 (dec512), the
+  /// permission will be passed during transfer.
+  int64_t permission{512};
```

Comment:
Why this check against 512? Can't the default value be 0 and wdt always populates it to the right value in DirectorySourceQueue?